### ISSUE TITLE
Wrong declaration AnchorPlacement type.

### ISSUE THREAD

**Issue by naikordian** [jsplumb/jsplumb#1057]
If I use anchor  type `number[]`  TS compiler will show a type error but everything ok when runtime:
![image](https://user-images.githubusercontent.com/45289300/132492126-512a19c5-cc77-4999-999f-7a131f73d74f.png)

![image](https://user-images.githubusercontent.com/45289300/132490765-90f9ef31-fb5e-4bd0-a801-21c92a5888be.png)
![image](https://user-images.githubusercontent.com/45289300/132489955-be215e8a-e8c8-4739-aefb-c046f6e19d1b.png)
If I user anchor type `AnchorPlacement `  TS compiler seem fine but give me a runtime error:
![image](https://user-images.githubusercontent.com/45289300/132490025-7f1b362d-6fbd-4174-a4c3-0266679cf075.png)
![image](https://user-images.githubusercontent.com/45289300/132490093-08b1b66e-71c6-4642-b747-7a0beb52a18a.png)

Version: 5.0.0-RC23

**Comment by sporritt** [comment]
AnchorPlacement is definitely not what you want, that's an internal interface. It surprises me that this compiles but I"m on my phone now and I cant investigate further until tomorrow.

**Comment by sporritt** [comment]
So this is the `addEndpoint` method signature:

https://docs.jsplumbtoolkit.com/community/apidocs/core.jsplumbinstance.addendpoint

```
addEndpoint(el: T["E"], params?: EndpointOptions<T["E"]>, referenceParams?: EndpointOptions<T["E"]>): Endpoint;
```

It takes the element, then some options and an optional second set of options, both of which are `EndpointOptions`:

https://docs.jsplumbtoolkit.com/community/apidocs/core.endpointoptions

`anchor` is an object of type `AnchorSpec`, which for some reason the api docs generator doesn't link from the `EndpointOptions` page, but its here:

https://docs.jsplumbtoolkit.com/community/apidocs/common.anchorspec

Following that down I see that `SingleAnchorSpec` does indeed support taking an `AnchorPlacement`:

https://docs.jsplumbtoolkit.com/community/apidocs/common.singleanchorspec

```
export declare type SingleAnchorSpec = AnchorId | FullAnchorSpec | AnchorPlacement | Array<AnchorPlacement>;
```

I'll have to look into this - it seems likely that if this method is called internally an anchor placement may be provided as argument. But it's definitely not something an API user should provide.

**Comment by sporritt** [comment]
@naikordian This should be fixed in the latest, 5.0.0-RC30.
